### PR TITLE
Tim i02 find pair

### PR DESCRIPTION
**Pull request by Timofey-Vorobev**
Implemented issue #02

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/php/vendor/
```
Comment (p5ych0) [inline .gitignore]: I believe it's already in .gitignore

File: php/Utils/Arrays.php [hunk 2]
```diff
@@ -20,7 +20,55 @@ class Arrays
      */
     public function findPair(int $sum)
     {
-        // @TODO: create body of the function and cover with tests
+        if ($sum < 0) {
```
Comment (p5ych0) [inline php/Utils/Arrays.php]: could be negative I believe

File: php/Utils/Arrays.php [hunk 3]
```diff
@@ -20,7 +20,55 @@ class Arrays
...
+        // no pairs found
+        return false;
+    }
+
+    /**
+     * Setter method for the internal array
+     * @param array $data
+     * @return $this
+     */
+    public function setValue(array $data)
```
Comment (p5ych0) [inline php/Utils/Arrays.php]: this is a bit incorrect way to assign value with regards to test.
use either Reflection or Mock

File: php/Utils/Arrays.php [hunk 4]
```diff
@@ -20,7 +20,55 @@ class Arrays
...
+    {
+        $this->array = $data;
+        $this->checkArray();
```
Comment (p5ych0) [inline php/Utils/Arrays.php]: also it's not correct to check data that must match for a certain method

File: php/Utils/Arrays.php [hunk 5]
```diff
@@ -20,7 +20,55 @@ class Arrays
...
+        return $this;
+    }
+
+
+    /**
+     * Throws an exception if array consists of not only numbers, or some of them are not whole
+     * @throws \DomainException
+     */
+    private function checkArray()
```
Comment (p5ych0) [inline php/Utils/Arrays.php]: an overhead

File: php/Utils/Arrays.php [hunk 6]
```diff
@@ -20,7 +20,55 @@ class Arrays
      */
     public function findPair(int $sum)
     {
-        // @TODO: create body of the function and cover with tests
+        if ($sum < 0) {
+            throw new \DomainException('Sum must be >= 0');
+        }
+        $result = [];
+        for ($i = 0; $i < count($this->array); $i++) {
+            if ($this->array[$i] > $sum) { // this is not a pair element as it is greater than the sum
```
Comment (p5ych0) [inline php/Utils/Arrays.php]: invalid condition check for the case where `$sum` is negative

File: php/Utils/Arrays.php [hunk 7]
```diff
@@ -20,7 +20,55 @@ class Arrays
...
+                continue;
+            }
+
+            $result[] = $this->array[$i];
+            $match = $sum - $this->array[$i];
+            for ($j = $i+1; $j < count($this->array); $j++) {
```
Comment (p5ych0) [inline php/Utils/Arrays.php]: please think about the complexety, currently it's O(nlog(n))

File: .gitignore (hunk 1 above)
Comment (Timofey-Vorobev) [reply .gitignore]: No, it was not. I made my own branch to start all commits from.
Also composer.lock

File: php/Utils/Arrays.php (hunk 2 above)
Comment (Timofey-Vorobev) [reply php/Utils/Arrays.php]: ok, this and other changes are pending,,,

File: .gitignore (hunk 1 above)
Comment (p5ych0) [reply .gitignore]: 1) look at `php/.gitignore`
2) composer.lock should not be ignored (check composer documentation)

**Pull request closed without merging**